xapian-applications/omega/omega.cc

```diff
@@ -447,17 +467,21 @@ try {
     // Percentage relevance cut-off
     val = cgi_params.find("THRESHOLD");
     if (val != cgi_params.end()) {
-	threshold = atoi(val->second.c_str());
-	if (threshold < 0) threshold = 0;
-	if (threshold > 100) threshold = 100;
+	unsigned int temp;
+	if (!parse_unsigned(val->second.c_str(),temp) || temp > 100) {
+	    throw "THRESHOLD parameter must be in the range 0-100\n";
```

Comment:
The documentation says (my emphasis):

> THRESHOLD
>        apply a percentage cut-off at the value given by this parameter
>        (**clipped to the range 0-100**).

Throwing an error here would make the actual behaviour inconsistent with the documented behaviour, and with what the actual behaviour has long been (since this feature was added in 2001 in d86ce4f19d2a4d7cfbadfde1531f383bd961e534).

Given this case is clearly documented, I think we need to continue to clip the value to 100 if it's too large.

Reply:
will fix it.

Reply:
It is also fairly logical that if you set a threshold to 110 for something which can only go up to 100 then that has exactly the same effect as setting it to 100.

If we were starting from a clean slate, then I think it would be reasonable to be strict about it, or to go for the current behaviour.